core/src/block.rs

```diff
@@ -248,23 +254,9 @@ impl<'x> OpenBlock<'x> {
 
     /// Turn this into a `LockedBlock`.
     pub fn close_and_lock(mut self, term_common_params: Option<&CommonParams>) -> Result<LockedBlock, Error> {
-        if let Err(e) = self.engine.on_close_block(&mut self.block, term_common_params) {
-            warn!("Encountered error on closing the block: {}", e);
-            return Err(e)
-        }
-        let header = self.block.header().clone();
-        for handler in self.engine.action_handlers() {
-            handler.on_close_block(self.block.state_mut(), &header).map_err(|e| {
-                warn!("Encountered error in {}::on_close_block", handler.name());
-                e
-            })?;
-        }
+        self.close_impl(term_common_params)?;
 
-        let state_root = self.block.state.commit().map_err(|e| {
-            warn!("Encountered error on state commit: {}", e);
-            e
-        })?;
-        if self.block.header.transactions_root() == &BLAKE_NULL_RLP {
+        if self.block.header.transactions_root().is_zero() || self.block.header.transactions_root() == &BLAKE_NULL_RLP {
```

Comment:
We don't need `self.block.header.transactions_root().is_zero()`.

Reply:
Sorry, I fixed it.